misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/ModelGenerator.java

```diff
@@ -39,7 +40,8 @@ public ModelGenerator(SemanticModel semanticModel, Module module) {
         this.module = module;
         Path moduleRootPath = module.project().sourceRoot().toAbsolutePath();
         if (module.moduleName().moduleNamePart() != null) {
-            moduleRootPath = moduleRootPath.resolve(module.moduleName().moduleNamePart());
+            moduleRootPath = moduleRootPath.resolve(ProjectDesignConstants.MODULES)
+                    .resolve(module.moduleName().moduleNamePart());
```

Comment:
have we tested this new logic against default module sources? cuz AFAIR `moduleNamePart` will not be null even for the default module. Isn't that the case?

Reply:
Yeah I tested the new logic with both default and non-default models, and it works as expected. and still the `moduleNamePart` is null for the default-module.

Reply:
great! 👍